### PR TITLE
[15.0][ADD] stock_product_set: New module

### PR DESCRIPTION
**Pull request by victoralmau**
New module

ejemplo

Please @pedrobaeza can you review it?

@Tecnativa TT46704

### CODE REVIEW

File: stock_product_set/README.rst [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+.. |badge4| image:: https://img.shields.io/badge/weblate-Translate%20me-F47D42.png
+    :target: https://translation.odoo-community.org/projects/stock-logistics-workflow-15-0/stock-logistics-workflow-15-0-stock_product_set
+    :alt: Translate me on Weblate
+.. |badge5| image:: https://img.shields.io/badge/runboat-Try%20me-875A7B.png
+    :target: https://runboat.odoo-community.org/builds?repo=OCA/stock-logistics-workflow&target_branch=15.0
+    :alt: Try me on Runboat
+
+|badge1| |badge2| |badge3| |badge4| |badge5|
+
+This module allows to use product sets in internal draft pickings.
```
Comment (pedrobaeza) [inline stock_product_set/README.rst]: It should be available in any type of pickings. The only requirement is that they are planned pickings.

File: stock_product_set/README.rst [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+Configuration
+=============
+
+#. Go to *Inventory > Configuration > Settings* and check "Storage Locations" option.
+
+Usage
+=====
+
+#. Go to *Inventory -> Internal transfer* and create new 'Planned transfer'.
```
Comment (pedrobaeza) [inline stock_product_set/README.rst]: Indicate how to create the "Planned transfer", as it's not obvious.

File: stock_product_set/README.rst [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+#. Go to *Add set* button and select some product set and click on the *Add set* button.
+#. Lines for each product of the set have been created in the *Stock Info* tab.
+
+Known issues / Roadmap
+======================
+
+The dependency sale_product_set must be changed to product_set in v16.
```
Comment (pedrobaeza) [inline stock_product_set/README.rst]:
```suggestion
* The dependency sale_product_set must be changed to product_set in v16.
```

File: stock_product_set/models/product_set_line.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+            "product_id": self.product_id.id,
+            "product_uom_qty": self.quantity * quantity,
+            "product_uom": self.product_id.uom_id.id,
+            # According to default values set from stock_move field in picking form
+            "location_id": picking.location_id.id,
+            "location_dest_id": picking.location_dest_id.id,
+            "group_id": picking.group_id.id,
+            "state": "draft",
+            "picking_id": picking.id,
+            "picking_type_id": picking.picking_type_id.id,
```
Comment (pedrobaeza) [inline stock_product_set/models/product_set_line.py:25]: You can use https://github.com/odoo/odoo/blob/260ef3dc73f5ed7d24f535145af01d7809dd7065/addons/purchase_stock/models/purchase.py#L506 as template.

I miss here at least `partner_id`.

File: stock_product_set/wizard/product_set_add_from_picking.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+            moves.append((0, 0, values))
+        return moves
+
+    def prepare_stock_move_data(self, set_line):
+        self.ensure_one()
+        line_values = set_line.prepare_picking_stock_move_values(
+            self.picking_id, self.quantity
+        )
+        sm_model = self.env["stock.move"]
+        line_values.update(sm_model.play_onchanges(line_values, line_values.keys()))
```
Comment (pedrobaeza) [inline stock_product_set/wizard/product_set_add_from_picking.py]: Don't do this. It's not needed here and you don't need the dependency on `onchange_helper` (although I know right now is "hidden"). Follow the same example put before in my comments from `purchase_stock`. Remove entirely this method and call directly the other prepare in set line.

File: stock_product_set/views/stock_picking_views.xml [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+        <field name="model">stock.picking</field>
+        <field name="inherit_id" ref="stock.view_picking_form" />
+        <field name="arch" type="xml">
+            <xpath expr="//header" position="inside">
+                <button
+                    class="btn-primary"
+                    type="action"
+                    name="%(act_open_wizard_product_set_add_from_picking)d"
+                    icon="fa-cubes"
+                    attrs="{'invisible': ['|', '|', ('picking_type_code','!=','internal'),('state', '!=', 'draft'), ('immediate_transfer', '=', True)]}"
```
Comment (pedrobaeza) [inline stock_product_set/views/stock_picking_views.xml]: Don't hide it for other picking type codes.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by victoralmau** [comment]
Changes done.

File: stock_product_set/README.rst [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+
+**Table of contents**
+
+.. contents::
+   :local:
+
+Configuration
+=============
+
+#. Go to *Inventory > Configuration > Settings* and check "Storage Locations" option.
```
Comment (pedrobaeza) [inline stock_product_set/README.rst]: This configuration is not needed at all.

File: stock_product_set/__manifest__.py [hunk 8]
```diff
@@ -0,0 +1,19 @@
+# Copyright 2023-2024 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+{
+    "name": "Stock Product Set",
```
Comment (pedrobaeza) [inline stock_product_set/__manifest__.py]:
```suggestion
    "name": "Add product sets in pickings",
```

File: stock_product_set/models/product_set_line.py [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+
+class ProductSetLine(models.Model):
+    _inherit = "product.set.line"
+
+    def prepare_picking_stock_move_values(self, picking, quantity):
+        self.ensure_one()
+        product = self.product_id.with_context(
+            lang=picking.partner_id.lang or self.env.user.lang
+        )
+        return {
```
Comment (pedrobaeza) [inline stock_product_set/models/product_set_line.py:15]:
```suggestion
        # Inspired in .... (and put the reference to Odoo code)
        return {
```

**Review by pedrobaeza — APPROVED** [review]

**Review by carolinafernandez-tecnativa — APPROVED** [review]
Funcional Review LGTM

**Comment by victoralmau** [comment]
It is ready to merge.

**Pull request merged**